lib/maths/CTimeSeriesDecompositionDetail.cc

```diff
@@ -472,13 +476,130 @@ std::size_t CTimeSeriesDecompositionDetail::CMediator::memoryUsage() const {
 
 //////// CPeriodicityTest ////////
 
+namespace {
+
+using TTimeTimeVecPrVec = std::vector<std::pair<core_t::TTime, TTimeVec>>;
+
+//! \brief Manages the choice of the tests' window parameters as a function
+//! of the job's bucket length.
+//!
+//! DESCRIPTION:\n
+//! The exact choice of window parameters is a tradeoff between the number
+//! of points used in the test and how quickly it finds periodic components.
+//! The fewer points the higher the chance of the false positives, but for
+//! long bucket lengths using many buckets means it takes a long time to
+//! find significant periodic components.
+class CPeriodicityTestWindowParameters {
+public:
+    static core_t::TTime maxBucketLength() { return 604800; }
+
+    static std::size_t numberBuckets(int window, core_t::TTime bucketLength) {
+        auto result = windowParameters(window, bucketLength);
+        return result != nullptr ? result->s_NumberBuckets : 0;
+    }
+
+    static const TTimeVec* bucketLengths(int window, core_t::TTime bucketLength) {
+        auto result = windowParameters(window, bucketLength);
+        return result != nullptr ? &result->s_BucketLengths : nullptr;
+    }
+
+    static const TTimeVec& testSchedule(int window, core_t::TTime bucketLength) {
+        return windowParameters(window, bucketLength)->s_TestSchedule;
+    }
+
+    static core_t::TTime shortestComponent(int window, core_t::TTime bucketLength) {
+        return windowParameters(window, bucketLength)->s_ShortestComponent;
+    }
+
+private:
+    struct CParameters {
```

Comment:
nit: `struct` names should begin with `S`

Reply:
See [93cbad3](https://github.com/elastic/ml-cpp/pull/1297/commits/93cbad37cbef56122803d5743d9ef6989eef8fd0).